lib/jormungandr/test/bench/Latency.hs

```diff
@@ -101,30 +111,106 @@ import Test.Integration.Framework.DSL
-        (wal1, wal2) <- twoFixtureWallet ctx
+    fmtLn "Latencies for 100 fixture wallets scenario"
+    runScenario logging tvar (nFixtureWallet 100)
+
+    fmtLn "Latencies for 2 fixture wallets with 10 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 10)
+
+    fmtLn "Latencies for 2 fixture wallets with 20 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 20)
+
+    fmtLn "Latencies for 2 fixture wallets with 100 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 100)
+
+    fmtLn "Latencies for 10 fixture wallets with 10 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 10)
+
+    fmtLn "Latencies for 10 fixture wallets with 20 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 20)
+
+    fmtLn "Latencies for 10 fixture wallets with 100 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 100)
+  where
+    nFixtureWallet n ctx = do
+        wal1 : wal2 : _ <- replicateM n (fixtureWallet ctx)
+        pure (wal1, wal2)
+
+    nFixtureWalletWithTxs n m ctx = do
+        (wal1, wal2) <- nFixtureWallet n ctx
+
+        let amt = (1 :: Natural)
+        let batchSize = 10
+        let whole10Rounds = div m batchSize
+        let lastBit = mod m batchSize
+        let amtExp val = fromIntegral ((fromIntegral val) + faucetAmt) :: Natural
+        let expInflows =
+                if whole10Rounds > 0 then
+                    [x*batchSize | x<-[1..whole10Rounds]] ++ [lastBit]
+                else
+                    [lastBit]
+        mapM_ (repeatPostTx ctx wal1 amt batchSize . amtExp) expInflows
+
+        pure (wal1, wal2)
+
+    repeatPostTx ctx wDest amtToSend batchSize amtExp = do
+        wSrc <- fixtureWallet ctx
+        let pass = "cardano-wallet" :: Text
+
+        replicateM_ batchSize (postTx ctx (wSrc, postTxEp, pass) wDest amtToSend)
+
+        rWal1 <- request @ApiWallet ctx (getWalletEp wDest) Default Empty
+        verify rWal1
+            [ expectSuccess
+            , expectEventually ctx getWalletEp balanceAvailable amtExp
+            ]
+
+        rDel <- request @ApiWallet ctx (deleteWalletEp wSrc) Default Empty
+        expectResponseCode @IO HTTP.status204 rDel
+
+        pure ()
+
+    postTx ctx (wSrc, postTxEndp, pass) wDest amt = do
+        addrs <- listAddresses ctx wDest
+        let destination = (addrs !! 1) ^. #id
+        let payload = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt},
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": #{pass}
+            }|]
+        r <- request @(ApiTransaction n) ctx (postTxEndp wSrc) Default payload
+        expectResponseCode HTTP.status202 r
+        return r
+
+    runScenario logging tvar scenario =
+        withUtf8Encoding $ benchWithServer logging $ \ctx -> do
```

Comment:
You can move `withUtf8Encoding` right to the top of `main`. It's one-time setup.

Reply:
done